packages/compass-aggregations/src/modules/is-pipeline-loaded-from-external.ts

```diff
@@ -0,0 +1,121 @@
+import type { Reducer } from 'redux';
+
+import type { NewPipelineConfirmedAction } from './is-new-pipeline-confirm';
+import { ActionTypes as ConfirmNewPipelineActions } from './is-new-pipeline-confirm';
+import { ActionTypes as PipelineModeActionTypes } from './pipeline-builder/pipeline-mode';
+import type { PipelineModeToggledAction } from './pipeline-builder/pipeline-mode';
+import type {
+  LoadGeneratedPipelineAction,
+  PipelineGeneratedFromQueryAction,
+} from './pipeline-builder/pipeline-ai';
+import { AIPipelineActionTypes } from './pipeline-builder/pipeline-ai';
+import type { RestorePipelineAction } from './saved-pipeline';
+import { RESTORE_PIPELINE } from './saved-pipeline';
+import { isAction } from '../utils/is-action';
+import type { EditorValueChangeAction } from './pipeline-builder/text-editor-pipeline';
+import { EditorActionTypes } from './pipeline-builder/text-editor-pipeline';
+import type {
+  ChangeStageDisabledAction,
+  ChangeStageOperatorAction,
+  ChangeStageValueAction,
+  StageAddAction,
+  StageMoveAction,
+  StageRemoveAction,
+} from './pipeline-builder/stage-editor';
+import { StageEditorActionTypes } from './pipeline-builder/stage-editor';
+
+export type NameState = string;
+
+export type IsPipelineLoadedFromExternal = {
+  // Used to visually indicate when the pipeline has been loaded
+  // from something like the ai generator or pipeline history.
+  isPipelineLoadedFromExternal: boolean;
+  pipelineLoadedFromExternalId: number;
+};
+
+const enum PipelineLoadedFromExternalActionTypes {
+  ClearIsPipelineLoadedFromExternal = 'compass-aggregations/ClearIsPipelineLoadedFromExternal',
+}
+
+type ClearIsPipelineLoadedFromExternalAction = {
+  type: PipelineLoadedFromExternalActionTypes.ClearIsPipelineLoadedFromExternal;
+};
+
+export function clearIsPipelineLoadedFromExternal(): ClearIsPipelineLoadedFromExternalAction {
+  return {
+    type: PipelineLoadedFromExternalActionTypes.ClearIsPipelineLoadedFromExternal,
+  };
+}
+
+export const INITIAL_STATE: IsPipelineLoadedFromExternal = {
+  isPipelineLoadedFromExternal: false,
+  pipelineLoadedFromExternalId: 0,
```

Comment:
Can this be also named `pipelineLoadedFromExternalKey`? This is the same thing, right?